README.md
# Divine_Card -  Flutter Package

Divine_Card provides a beautiful card with Images,buttons and tagline(text) stack upon it with beautiful offset shadow.

<img src = "banners.png" />

# How to install this package

  - **1. Depend on it**

    Add this to your flutter app's pubspec.yaml file:
    
    ```
    dependencies:
      divine_card: ^0.1.0
    ```
    
  - **2. Install it**
    
    You can install packages from the command line:

    with Flutter:
    
    ```
    $ flutter pub get
    ```
    
    Alternatively, your editor might support flutter pub get. Check the docs for your editor to learn more.
    
# How to use this package

  - **1. Import it**
  
    In your Dart code, import the package as mentioned below:
    
    ```
    import 'package:divine_card/divine_card.dart';
    ```
    
  - **2. Use It**
  
  ```
   divineCard(
     height: 200.0,
     color: Color(0xFF00578D),
     shadowColor: Colors.black38,
     blurRadius: 12.0,
     img: AssetImage('images/Picture2.png'),
     imgHeight: 200.0,
     imgWidth: 200.0,
     buttonText: 'Your Text',
     buttonColor: Colors.black,
     buttonBackground: Colors.white,
     icon: Icons.arrow_forward_ios,
     iconColor: Colors.black,
     title: 'Your text goes',
     subtitle: 'Here..',
     fontSize: 18.0,
     fontColor: Colors.white,
 
             ),
  ```

# About Me

Hi! my name is Radhika Goswami.

I'm 16 year Girl who get's amazed by seeing apps and wondered if I can too make these and just because of
Flutter now I can create APPs of my own !
 
[Visit this Repo](https://github.com/RadhikaGoswamiDivineCoding)

# Find me on

**[YouTube](https://www.youtube.com/channel/UC96JyllXkE4_Gfh9eLyaovw)**

**[Twitter](https://twitter.com/DivineCoding)**


**Email: radhika.divinecoding@gmail.com**

**Have a nice fluttery day! **

CHANGELOG.md
## [0.1.0] - 19/05/2020

*README updated.
*Example file added

lib/src/divine_card.dart
import 'package:flutter/material.dart';

// ignore: camel_case_types
class divineCard extends StatelessWidget {
  divineCard(
      {@required this.height,
        @required this.color,
        this.blurRadius,
        this.buttonColor,
        this.buttonText,
        this.subtitle,
        this.title,
        this.iconColor,
        this.fontColor,
        this.fontSize,
        this.icon,

        this.imgHeight,
        this.imgWidth,
        this.buttonBackground,
        this.shadowColor,
        @required this.img});
  final double height;
  final Color color;
  final Color shadowColor;
  final String title;
  final String subtitle;
  final double imgHeight;
  final double imgWidth;
  final double fontSize;
  final Color fontColor;
  final Color iconColor;
  final Color buttonColor;
  final double blurRadius;
  final String buttonText;
  final Color buttonBackground;
  final IconData icon;
  final ImageProvider img;

  @override

  Widget build(BuildContext context) {
    return Stack(
      children: <Widget>[
        Container(
          height: height,
          decoration: BoxDecoration(
              shape: BoxShape.rectangle,
              color: color,
              borderRadius: BorderRadius.circular(20.0),
              boxShadow: <BoxShadow>[
                BoxShadow(
                    color:shadowColor,
                    offset: Offset(0.0, 10.0),
                    blurRadius: blurRadius)
              ]),
        ),
        Container(
          alignment: FractionalOffset.centerRight,
          child: Image(
            image: img,
            height: imgHeight,
            width: imgWidth,
          ),
        ),
        Padding(
          padding: const EdgeInsets.only(left: 10.0, top: 30.0),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: <Widget>[
              Text(
                title,
                style: TextStyle(
                    color: fontColor,
                    fontSize: fontSize,
                    fontWeight: FontWeight.w600),
              ),
              SizedBox(
                height: 10.0,
              ),
              Text(
                subtitle,
                style: TextStyle(
                    color: fontColor,
                    fontSize: fontSize,
                    fontWeight: FontWeight.w600),
              ),
              SizedBox(
                height: 40.0,
              ),
              Container(
                height: 40.0,
                width: 150.0,
                decoration: BoxDecoration(
                  borderRadius: BorderRadius.circular(30),
                  color: buttonBackground,
                ),
                child: Padding(
                  padding: const EdgeInsets.only(left:15.0),
                  child: Row(
                    children: <Widget>[
                      Text(
                        buttonText,
                        style: TextStyle(
                          color: buttonColor,
                        ),
                      ),
                      SizedBox(
                        width: 20.0,
                      ),
                      Icon(
                        icon,
                      )
                    ],
                  ),
                ),),
            ],
          ),
        ),
      ],
    );
  }
}


example/example.dart
import 'package:divine_card/src/divinecard.dart';
import 'package:flutter/material.dart';


class HomePage extends StatefulWidget {
  @override
  _HomePageState createState() => _HomePageState();
}

class _HomePageState extends State<HomePage> with TickerProviderStateMixin {
  TabController _tabController;
  @override
  void initState() {
    _tabController = TabController(vsync: this, length: 3);
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.white,
      appBar: AppBar(
        leading: Icon(
          Icons.menu,
          color: Color(0xFF4F3C75),
          size: 30,
        ),
        elevation: 0.0,
        backgroundColor: Colors.white,
      ),
      body: Padding(
        padding: const EdgeInsets.only(left: 20.0, right: 20.0),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: <Widget>[
            Text(
              'Try DivineCard -',
              style: TextStyle(
                  color: Color(0xFF4F3C75),
                  fontSize: 25.0,
                  fontWeight: FontWeight.bold),
            ),
            Text(
              'Today.',
              style: TextStyle(
                  color: Color(0xFF4F3C75),
                  fontSize: 25.0,
                  fontWeight: FontWeight.bold),
            ),
            SizedBox(
              height: 20.0,
            ),
            TabBar(
              labelColor: Color(0xFF4F3C75),
              controller: _tabController,
              unselectedLabelColor: Color(0xFF4F3C75),
              isScrollable: true,
              indicatorColor: Color(0xFF4F3C75),
              tabs: <Widget>[
                Tab(
                  text: "Full list",
                ),
                Tab(
                  text: "Popular",
                ),
                Tab(
                  text: "Rare cases",
                ),
              ],
            ),
            SizedBox(
              height: 10.0,
            ),
            divineCard(
              height: 200.0,
              color: Color(0xFF00578D),
              shadowColor: Colors.black38,
              blurRadius: 12.0,
              img: AssetImage('images/Picture2.png'),
              imgHeight: 200.0,
              imgWidth: 200.0,
              buttonText: 'Your Text',
              buttonColor: Colors.black,
              buttonBackground: Colors.white,
              icon: Icons.arrow_forward_ios,
              iconColor: Colors.black,
              title: 'Your text goes',
              subtitle: 'Here..',
              fontSize: 18.0,
              fontColor: Colors.white,

            ),
          ],
        ),
      ),
    );
  }
}
